mirgecom/navierstokes.py

```diff
@@ -520,3 +537,283 @@ def ns_operator(dcoll, gas_model, state, boundaries, *, time=0.0,
+            dcoll, state, gas_model, boundaries, quadrature_tag,
+            dd=dd_vol, comm_tag=comm_tag)
+
+    vol_state_quad, inter_elem_bnd_states_quad, domain_bnd_states_quad = \
+        operator_states_quad
+
+    # {{{ Local utilities
+
+    # transfer trace pairs to quad grid, update pair dd
+    interp_to_surf_quad = partial(tracepair_with_discr_tag, dcoll, quadrature_tag)
+
+    # }}}
+
+    # {{{ === Compute grad(CV) ===
+
+    if grad_cv is None:
+        grad_cv = grad_cv_operator(
+            dcoll, gas_model, boundaries, state, time=time,
+            numerical_flux_func=gradient_numerical_flux_func,
+            quadrature_tag=quadrature_tag, dd=dd_vol,
+            operator_states_quad=operator_states_quad, comm_tag=comm_tag)
+
+    # Communicate grad(CV) and put it on the quadrature domain
+    grad_cv_interior_pairs = [
+        # Get the interior trace pairs onto the surface quadrature
+        # discretization (if any)
+        interp_to_surf_quad(tpair=tpair)
+        for tpair in interior_trace_pairs(
+            dcoll, grad_cv, volume_dd=dd_vol, comm_tag=(_NSGradCVTag, comm_tag))
+    ]
+
+    # }}} Compute grad(CV)
+
+    # {{{ === Compute grad(temperature) ===
+
+    if grad_t is None:
+        grad_t = grad_t_operator(
+            dcoll, gas_model, boundaries, state, time=time,
+            numerical_flux_func=gradient_numerical_flux_func,
+            quadrature_tag=quadrature_tag, dd=dd_vol,
+            operator_states_quad=operator_states_quad, comm_tag=comm_tag)
+
+    # Create the interior face trace pairs, perform MPI exchange, interp to quad
+    grad_t_interior_pairs = [
+        # Get the interior trace pairs onto the surface quadrature
+        # discretization (if any)
+        interp_to_surf_quad(tpair=tpair)
+        for tpair in interior_trace_pairs(
+            dcoll, grad_t, volume_dd=dd_vol,
+            comm_tag=(_NSGradTemperatureTag, comm_tag))
+    ]
+
+    # }}} compute grad(temperature)
+
+    # {{{ === Navier-Stokes RHS ===
+
+    # Physical viscous flux in the element volume
+    vol_term = viscous_flux(state=vol_state_quad,
+                     # Interpolate gradients to the quadrature grid
+                     grad_cv=op.project(dcoll, dd_vol, dd_vol_quad, grad_cv),
+                     grad_t=op.project(dcoll, dd_vol, dd_vol_quad, grad_t))
+
+    # Physical viscous flux (f .dot. n) is the boundary term for the div op
+    bnd_term = viscous_flux_on_element_boundary(
+        dcoll, gas_model, boundaries, inter_elem_bnd_states_quad,
+        domain_bnd_states_quad, grad_cv, grad_cv_interior_pairs,
+        grad_t, grad_t_interior_pairs, quadrature_tag=quadrature_tag,
+        numerical_flux_func=viscous_numerical_flux_func, time=time,
+        dd=dd_vol)
+
+    ns_rhs = \
+        (div_operator(dcoll, dd_vol_quad, dd_allfaces_quad, vol_term, bnd_term)
+         + inviscid_term)
+
+    if return_gradients:
+        return ns_rhs, grad_cv, grad_t
+    return ns_rhs
```

Comment:
(Same comments as for the Euler version.)

Reply:
This function has been removed completely.